pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -6,17 +6,61 @@
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order):
+def detect_minima(asig, order, interpolation_points, interpolation):
     signal = asig.as_array()
+    sampling_time = asig.times[1] - asig.times[0]
+
     t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
+    
+    if interpolation:
+
+        fitted_idx_times = np.zeros([len(t_idx)])
+        
+        start_arr = t_idx - int(interpolation_points/2)
+        start_arr = np.where(start_arr > 0, start_arr, 0)
+        stop_arr = start_arr + int(interpolation_points)
+        stop_arr = np.where(stop_arr < len(signal), stop_arr, len(signal))
+        
+        signal_arr = np.empty((interpolation_points,len(start_arr)))
+        signal_arr[:] = np.nan
+        
+        for i, (start, stop, channel_i) in enumerate(zip(start_arr, stop_arr, channel_idx)):
+            signal_arr[0:stop-start,i] = signal[start:stop, channel_i]
 
-    sort_idx = np.argsort(t_idx)
+        X_temp = range(0, interpolation_points)
+        params = np.polyfit(X_temp, signal_arr, 2)
+        # parabola translation
+        params[1,:] = params[1,:] - 2.*params[0,:]*start_arr
+        params[2,:] = params[2,:] -  params[0,:]*start_arr*start_arr - params[1,:]*start_arr
+        fitted_idx_times = -(1.*params[1,:])/(2.*params[0,:])*sampling_time
+        
+        # borders
+        idx = np.where(np.isnan(fitted_idx_times))[0]
+        print(idx)
+        while len(idx) > 0:
+            X_temp = range(0, interpolation_points - 1)
+            params = np.polyfit(X_temp, signal_arr[0:interpolation_points - 1, idx], 2)
+            params[1,:] = params[1,:] - 2.*params[0,:]*start_arr[idx]
+            params[2,:] = params[2,:] -  params[0,:]*start_arr[idx]*start_arr[idx] - params[1,:]*start_arr[idx]
+            fitted_idx_times[idx] = -(1.*params[1,:])/(2.*params[0,:])*sampling_time
+            idx = np.where(np.isnan(fitted_idx_times))[0]
```

Comment:
you get nan results because for minima near the end there are not the required number of interpolation points. The missing values are nan, and thus polyfit returns nan for these minima surroundings.
You can add
`start_arr = np.where(stop_arr < len(signal), start_arr, len(signal)-interpolation_points-1)`
before line 22, then both borders are handled the same and there shouldn't be nans in the output anymore.